Makefile.js

```diff
@@ -213,31 +213,35 @@ function generateRuleIndexPage(basedir) {
         categoryList = "conf/category-list.json",
         categoriesData = JSON.parse(cat(path.resolve(categoryList)));
 
-    find(path.join(basedir, "/lib/rules/")).filter(fileType("js")).forEach(filename => {
-        const rule = require(filename);
-        const basename = path.basename(filename, ".js");
-
-        if (rule.meta.deprecated) {
-            categoriesData.deprecated.rules.push({
-                name: basename,
-                replacedBy: rule.meta.docs.replacedBy || []
-            });
-        } else {
-            const output = {
+    find(path.join(basedir, "/lib/rules/")).filter(fileType("js"))
```

Comment:
Instead of looking through the filesystem here, it might be easier to use `require('.').linter.getRules()`.

```js
const rules = require(".").linter.getRules();

Array.from(rules.keys()).sort().forEach(ruleName => {
    const rule = rules.get(ruleName);

    // ...
});
```